scripts/build.js

```diff
@@ -1,35 +1,60 @@
-import { copyFile, mkdir } from "node:fs/promises";
+import { stat, copyFile, mkdir } from "node:fs/promises";
 import { pathToFileURL } from "node:url";
 
 import esbuild from "esbuild";
 
+const maybeLocalConfig = async (name = "local.config.js") => {
+  let exists = await stat(name)
+    .then(st => st.isFile())
+    .catch(err => {
+      if (err.code !== "ENOENT") throw err;
+    });
+  if (exists) {
+    console.log("Local config in use: %o", name);
+    return {
+      define: {
+        "CONFIG_PATH": JSON.stringify("../" + name),
+      },
+    }
+  }
+}
+
 const buildWith = async (key, buildOptions) => {
   await mkdir("dist", { recursive: true });
   await Promise.all(
     ["index.html", "shieldtest.html", "favicon.ico"].map((f) =>
       copyFile(`src/${f}`, `dist/${f}`)
     )
   );
-  return esbuild[key]({
+
+  const localConfig = await maybeLocalConfig();
+
+  const options = {
     entryPoints: ["src/americana.js", "src/shieldtest.js"],
     format: "esm",
     bundle: true,
     minify: true,
     sourcemap: true,
     outdir: "dist",
     logLevel: "info",
+    ...localConfig,
     ...buildOptions,
-  });
+    define: {
+      ...localConfig?.define,
+      ...buildOptions?.define,
+    },
+  };
+  return esbuild[key](options)
+    // esbuild will pretty-print its own error messages;
+    // suppress node.js from printing the exception.
+    .catch(() => process.exit(1));
```

Comment:
Note: moved this around to more carefully catch promise rejections from esbuild, but errors elsewhere in this function will not be swallowed, and _synchronous_ exception from esbuild.build will also throw if it doesn't like the options we passed.

Reply:
I wish I could say I understand how this works, but it certainly does work.